ampersand-state.js

```diff
@@ -177,11 +178,12 @@ assign(Base.prototype, Events, {
             }
 
             isEqual = this._getCompareForType(def.type);
+            onChange = this._getOnChangeForType(def.type);
             dataType = this._dataTypes[def.type];
 
             // check type if we have one
             if (dataType && dataType.set) {
-                cast = dataType.set(newVal);
+                cast = dataType.set.call(this, newVal, options);
```

Comment:
It's in the same vein as the `_getCompareForType` which binds `this`.  Useful if you need to do something with the state in the `onChange` function.  For example, the current 'state' datatype calls 'stopListening' on `this` when swapping out states.